src/plugins/intel_gpu/src/graph/concatenation.cpp

```diff
@@ -154,7 +154,7 @@ concatenation_inst::typed_primitive_inst(network& network, concatenation_node co
             auto nodes_list = stack.front();
             stack.pop_front();
 
-            for (auto processed_nodes : *nodes_list) {
+            for (auto& processed_nodes : *nodes_list) {
```

Comment:
May use a const ref in cases of changing copies to a reference in non-trivial sections of code. To avoid potential errors with data changes.